### PR TITLE
fix: throw io exception instead of storage exception

### LINKED ISSUE #219: BlobReadChannel read trows a non-IOException
Couldn't find this discussed anywhere and thought it belonged here, I hope you agree, please let me know otherwise.

[`BlobReadChannel.read`](https://github.com/googleapis/java-storage/blob/035ae2ad876fc9d17f86449d17f675000831b185/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java#L114) is annotated with `@throws IOException` which fits the interface and expected behaviour with nio (e.g. I get the expected `IOException` on read in the standard `ReadableBytesChannel` interface and in an input stream created with `Channels.newInputStream`)

However, at least one of the [current exceptions](https://github.com/googleapis/java-storage/blob/035ae2ad876fc9d17f86449d17f675000831b185/google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java#L136) thrown is `StorageException`, which is *not* an `IOException`, forcing users to know that this particular `ReadbleBytesChannel` throws another exception.

Could the exceptions thrown here be modified to inherit from `IOException` instead?

### PR DESCRIPTION
**Pull request by dmitry-fa**
Fixes #219

### CODE REVIEW

**Comment by frankyn** [comment]
@elharo could I get your review on this PR.

I want another perspective.

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java [hunk 1]
```diff
@@ -133,12 +133,12 @@ public int read(ByteBuffer byteBuffer) throws IOException {
...
           StringBuilder messageBuilder = new StringBuilder();
           messageBuilder.append("Blob ").append(blob).append(" was updated while reading");
-          throw new StorageException(0, messageBuilder.toString());
+          throw new IOException(messageBuilder.toString());
         }
         lastEtag = result.x();
         buffer = result.y();
       } catch (RetryHelper.RetryHelperException e) {
-        throw StorageException.translateAndThrow(e);
+        throw new IOException(e);
```
[the remark this answers is not in the document]
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java:141]: I don't see big difference for user in handling StorageException or IOException here. Both exception just wrap original cause and do not provide any extra information.

[the remark this answers is not in the document]
Comment (elharo) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java:141]: The big difference is that StorageException is a runtime exception and IOException is a checked exception. Anytime an exception arises from something outside the program itself such as an I/O error or a failed DNS lookup, a checked exception should be used. Anytime a problem arises due to a bug in a program itself--e.g. iterating past the end of an array--a runtime exception should be used.

[the remark this answers is not in the document]
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BlobReadChannel.java:141]: Yes, of course difference between checked and unchecked is big. I meant that after replacing StorageException with IOException users will not lose anything.

**Pull request merged**